### PR TITLE
Issue 95: Updated README.md

### PR DESCRIPTION
**Pull request by Aparnarr**
Signed-off-by: Aparna <Aparna.Rr@emc.com>

1. Content base on issue#87 included:

### CODE REVIEW

**Comment by Aparnarr** [comment]
@adrianmo 
1. Can i change the Version from 0.3.2 to 0.4.0??
2. In the `build operator image section`
        a. Should i remove `[operator-sdk]` from requirements??
        b. Shoul di remove operator-sdk from and replace with `make` in `$ operator-sdk build pravega/pravega-operator`
Need ur guidance

**Comment by adrianmo** [comment]
> Can i change the Version from 0.3.2 to 0.4.0??

No, 0.4.0 has not been released yet. 

> In the build operator image section
> a. Should i remove [operator-sdk] from requirements??

Yes.

> b. Should i remove operator-sdk from and replace with make in $ operator-sdk build pravega/pravega-operator

The new command to build the operator image is:

```
$ make build
```
That will generate a Docker image with the image tag with `latest` and also the following format `<latest_release_tag>-<number_of_commits_after_the_release>` + `-dirty` if there are uncommitted changes. 

Example image after running `make build`.

```
$ docker images
REPOSITORY                                                    TAG                             IMAGE ID            CREATED             
SIZE                                                                                                                                  
pravega/pravega-operator                                      0.1.1-3-dirty                   2b2d5bcbedf5        10 minutes ago      
41.7MB                                                                                                                                
pravega/pravega-operator                                      latest                          2b2d5bcbedf5        10 minutes ago
```

**Comment by Aparnarr** [comment]
@adrianmo  addressed all your comments. Will update in Zookeeper operator

File: README.md [hunk 1]
```diff
@@ -241,6 +243,7 @@ http://<cluster-name>-pravega-controller.<namespace>:10080/
 
 [Check this](#direct-access-to-the-cluster) to enable direct access to the cluster for development purposes.
 
+
```
[the remark this answers is not in the document]
Comment (Aparnarr) [reply README.md]: Accepted suggestion

File: README.md [hunk 2]
```diff
@@ -367,20 +370,27 @@ spec:
...
 Requirements:
   - Go 1.10+
-  - [Operator SDK](https://github.com/operator-framework/operator-sdk#quick-start)
 
-Use the `operator-sdk` command to build the Pravega operator image.
+Use the `make` command to build the Pravega operator image.
 
 ```
-$ operator-sdk build pravega/pravega-operator
+$ make build pravega/pravega-operator
```
[the remark this answers is not in the document]
Comment (Aparnarr) [reply README.md]: Accepted suggestion

File: README.md [hunk 3]
```diff
@@ -367,20 +370,27 @@ spec:
...
 ```
+That will generate a Docker image with the image tag with `latest` and also the following format
```
[the remark this answers is not in the document]
Comment (Aparnarr) [reply README.md]: Accepted suggestion

File: README.md [hunk 4]
```diff
@@ -367,20 +370,27 @@ spec:
...
+ `<latest_release_tag>-<number_of_commits_after_the_release> + -dirty`, if there are uncommitted changes.
+
+Example image after running make build.
```
[the remark this answers is not in the document]
Comment (Aparnarr) [reply README.md]: Accepted suggestion

**Comment by Aparnarr** [comment]
@adrianmo all comments addressed

File: README.md [hunk 5]
```diff
@@ -431,4 +442,34 @@ $ operator-sdk up local
...
 ## Releases  
 
-The latest pravega releases can be found on the [Github Release](https://github.com/pravega/pravega-operator/releases) project page.
+The latest Pravega releases can be found on the [Github Release](https://github.com/pravega/pravega-operator/releases) project page.
+
+## Troubleshooting
+
+### Helm Error: no available release name found
+
+This section would guide the users when installing nfs-server-provisioner, if they encounter any error on failure of `helm install`.
```
Comment (adrianmo) [inline README.md]: Replace this whole section with:

---

When installing a cluster for the first time using `kubeadm`, the initialization defaults to setting up RBAC controlled access, which messes with permissions needed by **Tiller** to do installations, scan for installed components, and so on. `helm init` works without issue, but `helm list`, `helm install` and other commands do not work.

```
$ helm install stable/nfs-server-provisioner
Error: no available release name found
```

The following workaround can be applied to resolve the issue:

1. Create a service account for the Tiller.

```
kubectl create serviceaccount --namespace kube-system tiller
```

2. Bind that service account to the `cluster-admin` ClusterRole.

```
kubectl create clusterrolebinding tiller-cluster-rule --clusterrole=cluster-admin --serviceaccount=kube-system:tiller
```

3. Add the service account to the Tiller deployment.

```
kubectl patch deploy --namespace kube-system tiller-deploy -p '{"spec":{"template":{"spec":{"serviceAccount":"tiller"}}}}'
```

The above commands should resolve the errors and `helm install` should work correctly.

File: README.md [hunk 6]
```diff
@@ -23,6 +23,8 @@ The project is currently alpha. While no breaking API changes are currently plan
     * [Direct Access to Cluster](#direct-access-to-the-cluster)
     * [Run the Operator Locally](#run-the-operator-locally)
 * [Releases](#releases)
+- [Troubleshooting](#troubleshooting)
```
[the remark this answers is not in the document]
Comment (Aparnarr) [reply README.md]: Accepted suggestion

File: README.md (hunk 5 above)
Comment (Aparnarr) [reply README.md]: Accepted suggestion

**Comment by Aparnarr** [comment]
@adrianmo All your comments addressed! Please check

File: README.md [hunk 7]
```diff
@@ -367,20 +369,29 @@ spec:
...
 ```
 $ docker images pravega/pravega-operator
-REPOSITORY                 TAG                 IMAGE ID            CREATED             SIZE
-pravega/pravega-operator   latest              625dab6fe470        54 seconds ago      37.2MB
+REPOSITORY                  TAG               IMAGE ID           CREATED             
+SIZE                                                                                                                                  
+pravega/pravega-operator    0.1.1-3-dirty      2b2d5bcbedf5        10 minutes ago      
+41.7MB                                                                                                                                
+pravega/pravega-operator    latest             2b2d5bcbedf5        10 minutes ago
+
```
[the remark this answers is not in the document]
Comment (adrianmo) [reply README.md]: yes, the size would be the same, 41.7MB.

[the remark this answers is not in the document]
Comment (Aparnarr) [reply README.md]: ok . Edited!

**Pull request merged**